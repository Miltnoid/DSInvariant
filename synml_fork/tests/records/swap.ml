type nat =
  | O
  | S of nat

let swap: {x:nat; y:nat} -> {x:nat; y:nat} |>
  { {x=0; y=1} => {x=1; y=0}
  | {x=1; y=0} => {x=0; y=1} } = ?
