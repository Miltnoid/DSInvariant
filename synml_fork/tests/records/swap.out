(* Generated Tue Mar  3 15:20:27 2015 *)
let swap : { x : nat; y : nat } -> { x : nat; y : nat } =
  fun (r1:{ x : nat; y : nat }) -> {x = r1.y; y = r1.x}
;;
